### PR TITLE
Removed various app_manager wrap functions

### PR DESCRIPTION
**Pull request by orangejenny**
## Technical Summary
Branched off of https://github.com/dimagi/commcare-hq/pull/32075, only need to review my commits.

Since @AddisonDunn went to the trouble of wrapping and re-saving all apps to finish up the `_entered` convention work, most of the `wrap` functions in `app_manager.models` can also be removed. This PR removes wrap logic that manipulates the json doc before calling the super's wrap. It leaves wrap logic that differentiates between subclasses (FormBase and ModuleBase) and that manipulates the object after it's been wrapped.

Almost of the changes update test apps. I started by wrapping each test app, then calling to_json and re-saving it, which broke a lot of tests. Rather than play whack a mole, I went back and did it more methodically, removing smaller pieces of logic and running a script to update just that part of each test app.

This is kind of terrible to review because it's largely automated changes. If anyone would prefer to review as a whole, [here's the diff with the base branch](https://github.com/dimagi/commcare-hq/compare/ad/remove-entered-wrap-functions..jls/remove-a-lot-of-wrapping?w=1).

### Safety story
This is largely changes to test files.

### Automated test coverage

In PR.

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-4758

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
don't think tests are running, going to try this again

File: corehq/apps/app_manager/management/commands/wrap_test_apps.py [hunk 1]
```diff
@@ -167,6 +168,19 @@ def handle(self, **options):
...
+            # ApplicationBase.wrap
+            if 'build_langs' in doc:
+                if doc['build_langs'] != doc['langs'] and 'build_profiles' not in doc:
+                    doc['build_profiles'] = {
+                        uuid.uuid4().hex: dict(
+                            name=', '.join(doc['build_langs']),
+                            langs=doc['build_langs']
+                        )
+                    }
+                    should_save = True
```
Comment (snopoke) [inline corehq/apps/app_manager/management/commands/wrap_test_apps.py]:
```suggestion
                    should_write = True
```

**Review by snopoke — APPROVED** [review]

**Comment by biyeun** [comment]
@orangejenny there was an issue with one of Addison's remove wrap branches on staging causing the Case Data Export workflow to fail. Please see https://dimagi-dev.atlassian.net/browse/QA-4512

**Comment by orangejenny** [comment]
Realized this doesn't really depend on https://github.com/dimagi/commcare-hq/pull/32075/ ...but the two PRs do conflict, so I'll leave it as-is and wait to merge until that PR is ready and the issue @biyeun linked is resolved, just in case.

**Comment by esoergel** [comment]
Egads!
image
nice work

**Comment by orangejenny** [comment]
Realized this PR removes the last usage of `PropertyXpathGenerator`, so I removed that in 1ab8cfc3f5d320d40d3be05bb279779ec1a2b2bb

**Comment by orangejenny** [comment]
> Realized this PR removes the last usage of PropertyXpathGenerator, so I removed that in https://github.com/dimagi/commcare-hq/commit/1ab8cfc3f5d320d40d3be05bb279779ec1a2b2bb

Grep test wasn't enough here: [this decorator](https://github.com/dimagi/commcare-hq/blob/74e3f92af04937f771b097d10d468cf1ad57e31e/corehq/apps/app_manager/detail_screen.py#L575) combined with [this code](https://github.com/dimagi/commcare-hq/blob/74e3f92af04937f771b097d10d468cf1ad57e31e/corehq/apps/app_manager/detail_screen.py#L53) mean `PropertyXpathGenerator` is still used. It's now restored.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/app_manager/tests/test_filters.py [hunk 2]
```diff
@@ -328,3 +328,12 @@ def test_ancestor_location_type_filter_with_no_location_found(self):
         ui_filter = self._get_dynamic_choice_list_ui_filter()
         ancestor_state_value = ucr_filter.get_filter_value(self.nate, ui_filter)
         self.assertEqual(ancestor_state_value, ui_filter.default_value())
+
+
+class NumericFilterTests(SimpleTestCase):
+
+    def test_numeric_filter(self):
```
Comment (snopoke) [inline corehq/apps/app_manager/tests/test_filters.py:335]: 👍🏻

**Pull request merged**